azure/durable_functions/tasks/create_timer.py

```diff
@@ -0,0 +1,42 @@
+from typing import List
+from ..models.actions.CreateTimerAction import CreateTimerAction
+from ..models.history import HistoryEvent
+from .task_utilities import \
+    find_task_retry_timer_created, find_task_retry_timer_fired, set_processed
+import datetime
+from .timer_task import TimerTask
+
+
+def create_timer_task(state: List[HistoryEvent],
+                      fire_at: datetime) -> TimerTask:
+    """Durable Timers are used in orchestrator function to implement delays.
+
+    Parameters
+    ----------
+    state : List[HistoryEvent]
+        The list of history events to search to determine the current state of the activity
+    fire_at : datetime
+        The time interval to fire the timer trigger
+
+    Returns
+    -------
+    TimerTask
+        A Durable Timer Task that schedules the timer to wake up the activity
+    """
+    new_action = CreateTimerAction(fire_at)
+
+    timer_created = find_task_retry_timer_created(state, fire_at)
```

Comment:
`fire_at` makes `find_task_retry_timer` take as input a `datetime` object instead of an event object. I believe this will error out in the body of `find_task_retry_timer`. More details on my comment for this method :)

Reply:
Actually, just realized that this method was not modified as part of this PR, so I'll just include the details here:

So `find_task_retry_timer` contains the following expression:

```
    tasks = [e for e in state if e.event_type == HistoryEventType.TIMER_CREATED
             and e.event_id == failed_task.TaskScheduledId + 1]
```

where `failed_task` corresponds to the `fire_at` parameter. Since `datetime` does not contain a `TaskScheduledId` field, the expression will error out.

Reply:
I think the fix here is to change that array to something like:
```python
    tasks = [e for e in state if e.event_type == HistoryEventType.TIMER_CREATED
             and hasattr(e, "FireAt")
             and datetime.strptime(e.FireAt) == failed_task]
            # or simply cast failed_task to string and to the comparison that way, unsure which
            # is safer 
```

I got this from:  https://github.com/Azure/azure-functions-durable-js/blob/ea5b8561fe1280c19fcf01c72983673dd402569b/src/orchestrator.ts#L779

Reply:
find_task_retry _timer is reused by callactivity with retry that doesn't have FireAt. Maybe need a boolean to sort this out.

Reply:
Maybe the JS implementation has a solution to this? I don't know, perhaps Python diverged from JS in the implementation of this method

Reply:
Finally this is what I did:
e.FireAt == failed_task.strftime(DATETIME_STRING_FORMAT) to ensure a string comparison